src/Parser/Desugar.ml

```diff
@@ -708,7 +708,11 @@ and tr_def ?(public=false) (def : Raw.def) =
     let (pub_type, pub_ctors) =
       match vis with
       | DV_Private  -> (public, public)
-      | DV_Abstract -> (true,   public)
+      | DV_Abstract -> if public then
```

Comment:
Code formatting could be improved here. If a body of a match clause does not fit into one line, we break the line after arrow `->`. Moreover, the expression after `then` should have bigger indentation.